tests/tests.js

```diff
@@ -283,64 +284,61 @@ var testSuite = function () {
         $( '#select3' ).val( 'bar' );
-        $( '#select3' ).trigger( 'change', function () {
+        $( '#select3' ).one( 'change', function () {
           expect( garlicStorage.get( $( '#select3' ).garlic( 'getPath' ) ) ).to.be( 'bar' );
-        } )
+        } ).trigger( 'change' )
       } )
       it ( 'If radio button is selected, value or new value should be stored', function () {
         $( '#radio1' ).val( 'foo' );
-        $( '#radio1' ).trigger( 'change', function () {
+        $( '#radio1' ).one( 'change', function () {
           expect( garlicStorage.get( $( '#radio1' ).garlic( 'getPath' ) ) ).to.be( 'foo' );
-        } )
+        } ).trigger( 'change' )
       } )
       it ( 'Same, but with radio buttons not at the same DOM level', function () {
         $( '#radio2' ).val( 'bar' );
-        $( '#radio2' ).trigger( 'change', function () {
+        $( '#radio2' ).one( 'change', function () {
           expect( garlicStorage.get( $( '#radio2' ).garlic( 'getPath' ) ) ).to.be( 'bar' );
-        } )
+        } ).trigger( 'change' )
       } )
       it ( 'If a checkbox is checked, its state should be persisted', function () {
         $( '#checkbox4' ).val( 'foo' );
         $( '#checkbox6' ).val( 'bar' );
-        $( '#checkbox4' ).trigger( 'change', function () {
-          expect( garlicStorage.get( $( '#checkbox4' ).garlic( 'getPath' ) ) ).to.be( 'foo' );
-        } )
-        $( '#checkbox6' ).trigger( 'change', function () {
-          expect( garlicStorage.get( $( '#checkbox6' ).garlic( 'getPath' ) ) ).to.be( 'bar' );
-        } )
+        $( '#checkbox4' ).one( 'change', function () {
+          expect( garlicStorage.get( $( '#checkbox4' ).garlic( 'getPath' ) ) ).to.be( 'unchecked' );
+        } ).trigger( 'change' )
+        $( '#checkbox6' ).one( 'change', function () {
+          expect( garlicStorage.get( $( '#checkbox6' ).garlic( 'getPath' ) ) ).to.be( 'unchecked' );
+        } ).trigger( 'change' )
+      } )
+      it ( 'If a checkbox is checked, its state should be persisted', function () {
+        $( '#checkbox4' ).val( 'foo' );
+        $( '#checkbox4' ).prop("checked", true)
+        $( '#checkbox4' ).one( 'change', function () {
+          expect( garlicStorage.get( $( '#checkbox4' ).garlic( 'getPath' ) ) ).to.be( 'checked' );
+        } ).trigger( 'change' )
       } )
       it ( 'Same, but with checkboxes not at the same DOM level', function () {
         $( '#checkbox7' ).val( 'bar' );
         $( '#checkbox8' ).val( 'baz' );
-        $( '#checkbox7' ).trigger( 'change', function () {
-          expect( garlicStorage.get( $( '#checkbox7' ).garlic( 'getPath' ) ) ).to.be( 'bar' );
-        } )
-        $( '#checkbox8' ).trigger( 'change', function () {
-          expect( garlicStorage.get( $( '#checkbox8' ).garlic( 'getPath' ) ) ).to.be( 'baz' );
-        } )
-      } )
-      it ( 'If a checkbox is unchecked, its state should be removed from storage', function () {
```

Comment:
It looks like the current version of garlic.js doesn't honour this functionality. I removed this test and added the "If a checkbox is checked, its state should be persisted" test.